Feat/add dry run docs

zihanKuang opened the pull request:
This PR adds a page for the “Dry Run” functionality in Meshery's design workflows. The guide explains what Dry Run does, how to use it, and how it differs from Validate and Deploy.

### What's Included
- Supporting images (GIF + PNGs) for UI walkthroughs
- Usage explanation with screenshots
- Q&A and footnotes for known UI behaviors
- The section _“How It Works”_ reflects the real execution flow based on these implementations:
https://github.com/meshery/meshery/blob/master/ui/machines/validator/designValidator.js
https://github.com/meshery/meshery/blob/ec5eea1a098b12afdafe93cf1b821da99a3f6bfc/server/handlers/design_engine_handler.go
https://github.com/meshery/meshery/blob/72ad171fa7ae0975d3ffa57b7b27da37e4be70c0/server/models/pattern/patterns/k8s/k8s.go

- [x] Yes, I signed my commits.

hortison commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+categories: [Designer]
+tags: [designs]
+aliases:
+  - /meshmap/tasks/designs/dry-running-designs
```
If this is a brand-new page, there's no need for this baggage (for this redirect).

hortison commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+  - /meshmap/tasks/designs/dry-running-designs
+---
+
+## 1. Overview
```
This screams "you are now reading AI-generated content. Enjoy". Perhaps, we can do without?

hortison commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+## 1. Overview
+
+Dry Run lets you try out your design without making any changes to your Kubernetes clusters.  
+It’s a safe way to check if everything is set up correctly before you actually deploy.
```
Nice explanation. This is relatable and understandable.

hortison commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+
+![Find Dry Run Menu](/kanvas/tasks/designs/images/find-dry-run-menu.gif)
+
+## 2. Why Use Dry Run?
```
```suggestion
## Why you should perform Dry Runs
```
Just a suggestion... please reject at-will.

hortison commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+
+### 3.1 Using the Actions Dropdown Menu
+
+- In the [kanvas](https://playground.meshery.io/extension/meshmap), click the **Actions** dropdown.  
```
It's not "the kanvas". It is:

1. Kanvas, which is a pronoun, not a noun (e.g. not "the Zihan", just "Zihan").
2. Kanvas, which is a pronoun, not a non (e.g. not "zihan", but "Zihan"), because she is unique, special, and deserves capitalization.

leecalcote commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+
+![Actions menu showing Dry Run option](/kanvas/tasks/designs/images/actions-dropdown-menu.png)
+
+- After clicking, the results will appear:
```
```suggestion
- After clicking, results of your simulated deployment will appear:
```

leecalcote commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+
+## 6. Best Practices & Tips
+
+- Always **Validate → Dry Run → Deploy** in that order.  
```
```suggestion
- As a best practice, always **Validate → Dry Run → Deploy** - in that order.  
```

leecalcote commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+Dry Run does **not** include schema validation.  
+We recommend clicking **Validate first**, then **Dry Run**, then **Deploy**.
+
+## 6. Best Practices & Tips
```
Perhaps, this should be a numbered list?

leecalcote commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,99 @@
+- Use Dry Run in your CI pipelines to catch issues before merge.  
+- Ensure your Kubernetes context is correctly configured and you have network access to the clusters you are targeting.  
+
+[^1]: In the current release, you may occasionally encounter a case where the Arrow Dropdown Menu does not respond as expected. This is a known issue and is currently being addressed. In the meantime, please use the Actions Dropdown Menu as an alternative for triggering a Dry Run.
```
There's huge value in your discovery and raising of bugs, @zihanKuang. Is there an open issue on this behavior?

zihanKuang commented:
Hi @leecalcote @hortison, thanks a lot for the helpful feedback!

I've updated the PR based on your suggestions:
- Removed the unnecessary alias.
- Reworded headings and adjusted the tone for a more natural, user-focused voice.
- Converted the Best Practices section into a numbered list.

Also, I’ve opened an issue to track the dropdown behavior mentioned in the footnote: [#14584](https://github.com/meshery/meshery/issues/14584)

Let me know if there's anything else I can improve.

Paladian78 commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,95 @@
+
+> Dry Run does not create, update, or delete any live resources.
+
+## How to Perform a Dry Run
+
+In Kanvas, there are two ways to start a Dry Run:
+
+### Using the Actions Dropdown Menu
+
+- In [kanvas](https://playground.meshery.io/extension/meshmap), click the **Actions** dropdown.  
+- Select **Dry Run** from the list.
+
+![Actions menu showing Dry Run option](/kanvas/tasks/designs/images/actions-dropdown-menu.png)
+
+- After clicking, results of your simulated deployment will appear:
+
+![Actions menu next result ](/kanvas/tasks/designs/images/actions-dropdown-result.png)
+
+### Using the Arrow Dropdown Menu
+
+- At the top-right corner of the design mode in [kanvas](https://playground.meshery.io/extension/meshmap), click the **arrow-shaped dropdown button**. 
+- Select **Dry Run** from the list.
+
+![Dropdown Menu Shortcut](/kanvas/tasks/designs/images/arrow-dropdown-menu.png)
+
+- The simulation will be triggered immediately based on the clusters selected in your current context.
+- You will receive feedback via the **[Notification Center](https://docs.meshery.io/guides/infrastructure-management/notification-management)**, where you can view results for each component[^1].
+
```
I noticed that the "Actions" and the "Arrow" dropdown next to it seem to serve a similar purpose. Both options lead to the same set of steps, with one displaying just icons and the other including the heading along with the icon. Would it be possible to merge this into a single option—for example, Users can click either "Actions" or the Arrow dropdown to access features like Dry Run? This could help avoid repetition without any effect on the understanding of docs.

pontusringblom commented:
Thanks for reviewing, @Paladian78

leecalcote commented:
Regarding the difference in behavior on the Actions button - - https://cloud.layer5.io/academy/learning-paths/mastering-meshery/introduction-to-meshery?chapter=deploying-meshery-designs

The single-click action offers away for experienced users to bypass a wizard and simply directly invoke the selected action.

zihanKuang commented:
We'll hold off on further doc updates to this section until [issue #14634](https://github.com/meshery/meshery/issues/14634) is resolved 😎

leecalcote commented on content/en/kanvas/tasks/designs/dry-running-designs.md:
```diff
@@ -0,0 +1,95 @@
+
+### To Start a Dry Run
+
+1. Navigate to [Kanvas](https://playground.meshery.io/extension/meshmap).
```
Since we're describing Kanvas, let's use https://kanvas.new as the domain in the URLs, not playground.meshery....
